x-pack/plugins/security_solution/public/types.ts

```diff
@@ -46,8 +47,9 @@ export type StartServices = CoreStart &
     storage: Storage;
   };
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface PluginSetup {}
+export interface PluginSetup {
+  resolver: () => Promise<ResolverPluginSetup>;
```

Comment:
@XavierM I needed to make this async, otherwise all of Resolver is pulled into securitySolution's page load bundle.